pyproject.toml

```diff
@@ -0,0 +1,66 @@
+# NOTE: you have to use single-quoted strings in TOML for regular expressions.
+# It's the equivalent of r-strings in Python.  Multiline strings are treated as
+# verbose regular expressions by Black.  Use [ ] to denote a significant space
+# character.
+
+[tool.black]
+line-length = 88
+target-version = ['py310', 'py311']
+include = '\.pyi?$'
+exclude = '''
+/(
+    \.eggs
+  | \.git
+  | \.hg
+  | \.mypy_cache
+  | \.tox
+  | \.venv
+  | _build
+  | buck-out
+  | build
+  | dist
+  | migrations
+)/
+'''
+
+[tool.isort]
+use_parentheses=true
+multi_line_output=3
+include_trailing_comma=true
+line_length=79
+known_first_party=['task_processor']
+known_third_party=['django', 'rest_framework', 'saml2', 'drf_yasg2', 'pytest']
+skip = ['migrations', 'flagsmith', '.venv']
+
+[tool.poetry]
+name = "flagsmith_task_processor"
+version = "1.0.0"
+description = "Task Processor plugin for Flagsmith application."
+authors = ["Flagsmith <support@flagsmith.com>"]
+readme = "readme.md"
+
+[tool.poetry.dependencies]
+python = "^3.10"
+django = "~3.2.23"
```

Comment:
This is a really good point. I'm going to make the requirements much more relaxed here, but I will have to limit to <4.0 which will eventually cause us some pain when we upgrade to django 4.0.

Reply:
Note that I tried to extend the support to <5.0.0 to cover us, but unfortunately there is a breaking change in that we will need to drop the dependence on the addition `django-capture-on-commit-callbacks` library since this is part of the core django lib from 4.0.0+. poetry doesn't (as far as I can tell) support conditional dependencies on other libraries, only on the python version (see docs [here](https://python-poetry.org/docs/dependency-specification/#multiple-constraints-dependencies))

Reply:
Interesting. I hope the upgrade goes smoothly once we're ready.